### ISSUE TITLE
Unstable 4.19 and main branch smoketests

### ISSUE THREAD

**Issue by rohityadavcloud** [apache/cloudstack#8931]
Smoketests seen failing both in GitHub Actions jobs and in BO/smoketests in latest 4.19 and main branches.

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
GitHub Actions failures, Smoketests failures
~~~

**Comment by weizhouapache** [comment]
it looks an issue with codecov account
should we talk to apache infra ro codecov ?

@rohityadavcloud @DaanHoogland

**Comment by DaanHoogland** [comment]
this set of failures in smoke tests have become common in the last week(s): https://github.com/apache/cloudstack/pull/8839#issuecomment-2062838034

**Comment by weizhouapache** [comment]
it looks the issue has been addressed in #8930 which fixes a regression in #8800
 @vishesh92 may confirm it

**Comment by DaanHoogland** [comment]
> it looks an issue with codecov account should we talk to apache infra ro codecov ?
> 
> @rohityadavcloud @DaanHoogland

created https://issues.apache.org/jira/browse/INFRA-25725

**Comment by DaanHoogland** [comment]
another set of tests that fails at times is [vm lifecycle](https://github.com/apache/cloudstack/pull/8932#issuecomment-2065351549)

**Comment by weizhouapache** [comment]
> another set of tests that fails at times is [vm lifecycle](https://github.com/apache/cloudstack/pull/8932#issuecomment-2065351549)

@DaanHoogland 
yes, I am checking
looks like intermittent failures.

**Comment by DaanHoogland** [comment]
> > another set of tests that fails at times is [vm lifecycle](https://github.com/apache/cloudstack/pull/8932#issuecomment-2065351549)
> 
> @DaanHoogland yes, I am checking looks like intermittent failures.

It is not just your PR, @weizhouapache .  I've seen this a lot lately.

**Comment by weizhouapache** [comment]
> > > another set of tests that fails at times is [vm lifecycle](https://github.com/apache/cloudstack/pull/8932#issuecomment-2065351549)
> > 
> > @DaanHoogland yes, I am checking looks like intermittent failures.
> 
> It is not just your PR, @weizhouapache .  I've seen this a lot lately.

@DaanHoogland 
I've found a potential issue which might cause it.
let me fix it

**Comment by weizhouapache** [comment]
hope #8948 helps

**Comment by rohityadavcloud** [comment]
Let's do a new round of smoketests and close this if branches are stable enough cc @sureshanaparti @weizhouapache @DaanHoogland 

4.18 - https://github.com/apache/cloudstack/pull/7345
4.19 - https://github.com/apache/cloudstack/pull/8601
main/4.20 -  https://github.com/apache/cloudstack/pull/8602

**Comment by weizhouapache** [comment]
it looks the only issue now is

test_01_events_resource	Error	423.26	test_events_resource.py

**Comment by DaanHoogland** [comment]
yes a new one, test_event_resource.py started failing in several PR smoke test runs:
https://github.com/apache/cloudstack/pull/8889#issuecomment-2075766888
https://github.com/apache/cloudstack/pull/8891#issuecomment-2075472394
https://github.com/apache/cloudstack/pull/8809#issuecomment-2075900287
https://github.com/apache/cloudstack/pull/8937#issuecomment-2076084264

**Comment by weizhouapache** [comment]
I updated the integration test
```
diff --git a/test/integration/smoke/test_events_resource.py b/test/integration/smoke/test_events_resource.py
index 660cbd37bce..4bbab66532f 100644
--- a/test/integration/smoke/test_events_resource.py
+++ b/test/integration/smoke/test_events_resource.py
@@ -16,7 +16,6 @@
 # under the License.
 """ BVT tests for Events Resource
 """
-import json
 import os
 import tempfile
 import time
@@ -185,7 +184,7 @@ class TestEventsResource(cloudstackTestCase):
         for event in events:
             if event.type.startswith("VM.") or (event.type.startswith("NETWORK.") and not event.type.startswith("NETWORK.ELEMENT")) or event.type.startswith("VOLUME.") or event.type.startswith("ACCOUNT.") or event.type.startswith("DOMAIN.") or event.type.startswith("TEMPLATE."):
                 if event.resourceid is None or event.resourcetype is None:
-                    self.debug("Failed event:: %s" % json.dumps(event, indent=2))
+                    self.debug("Failed event:: %s" % event)
                     self.fail("resourceid or resourcetype for the event not found!")
                 else:
                     self.debug("Event %s at %s:: Resource Type: %s, Resource ID: %s" % (event.type, event.created, event.resourcetype, event.resourceid))
```

and got the following error (`resourceid` is not found)
`
test_01_events_resource (tests.smoke.test_events_resource.TestEventsResource): DEBUG: Failed event:: {id : '4a7bbf38-35ee-42d3-ada6-09ffc20b702c', username : 'admin', type : 'VOLUME.DESTROY', level : 'INFO', description : 'Successfully completed destroying a volume', account : 'admin', domainid : '37ff2485-cc1e-11ee-a4eb-1e00db000197', domain : 'ROOT', resourcetype : 'Volume', created : '2024-04-26T13:24:11+0000', state : 'Completed'}
`

it looks the error `test_01_events_resource` is caused by #8800  (merged on 2024-04-12)

cc @vishesh92

**Comment by DaanHoogland** [comment]
test_safe_shutdown seems the only one remaining

**Comment by sureshanaparti** [comment]
test_safe_shutdown fixed in #9276, failure not seen in recent tests.
